action.php

```diff
@@ -42,7 +42,7 @@ function start(&$event, $param) {
     function _load() {
         global $conf;
         // determine the path to the data
-        $userFile = $conf['savedir'] . '/virtualgrp.php';
+        $userFile = DOKU_INC . $conf['savedir'] . '/virtualgrp.php';
```

Comment:
This won't work if savedir is set to an absolute path. The plugin should probably be changed to store the configuration in the conf directory (DOKU_CONF) anyway.

Reply:
Ok, it never ocurred to my that an absolute path for `savedir` was possible. Taking that into account, I agree with the request. I must admit that I've also wondered about the location of `virtualgrp.php` myself, and believe it belongs in the configuration directory.

How about this plan of attack:

 1. first try reading/writing from `DOKU_CONF . /virtualgrp.php
 2. if reading from that location fails, try to copy `virtualgrp.php` from `$conf['savedir'] to `DOKU_CONF` (and if copy was successful, delete original location (ignoring failure to delete))
 3. if copying fails, try reading from `$conf['savedir'] as before

While I'm at it, would it be preferable to switch file format and name (at the new location) to `virtualgrp.json`, or should we keep it as it is?

Reply:
Sounds exactly like what I had in mind :+1:

Reply:
JSON would also be a good idea instead of using PHP's serialize.